File: src/main/java/com/github/thaynarasilvapinto/SimuladorBanco/controller/ClienteController.java
package com.github.thaynarasilvapinto.SimuladorBanco.controller;

import com.github.thaynarasilvapinto.SimuladorBanco.controller.response.ClienteResponse;
import com.github.thaynarasilvapinto.SimuladorBanco.domain.Cliente;
import com.github.thaynarasilvapinto.SimuladorBanco.domain.Conta;
import com.github.thaynarasilvapinto.SimuladorBanco.domain.ValidaCPF;
import com.github.thaynarasilvapinto.SimuladorBanco.services.ClienteService;
import com.github.thaynarasilvapinto.SimuladorBanco.services.ContaService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;



@RestController
@RequestMapping(value = "/")
public class ClienteController {
    @Autowired
    private ClienteService serviceCliente;
    @Autowired
    private ContaService serviceConta;

    @RequestMapping(value = "/cliente/{id}", method = RequestMethod.GET)
    public ResponseEntity<ClienteResponse> mostrarCliente(@PathVariable Integer id) {
        Cliente cliente = serviceCliente.find(id);
        if (cliente != null) {
            return ResponseEntity.ok().body(new ClienteResponse(cliente));
        }
        return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY).body(null);
    }

    @RequestMapping(value = "/criar-cliente", method = RequestMethod.POST)
    public ResponseEntity<ClienteResponse> criarCliente(@RequestBody Cliente c) {

        if(serviceCliente.findCPF(c.getCpf()) == null && new ValidaCPF().validaCPF(c.getCpf()) == true) {
            Cliente cliente = new Cliente(c.getNome(), c.getCpf(), null);
            Cliente clienteInserido = serviceCliente.insert(cliente);
            if (clienteInserido != null) {
                Conta conta = new Conta(0.00);
                Conta contaInserida = serviceConta.insert(conta);
                if (contaInserida != null) {
                    clienteInserido.setConta(contaInserida);
                    serviceCliente.update(clienteInserido);
                    return ResponseEntity.ok().body(new ClienteResponse(clienteInserido));
                }
            }
        }
        return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY).body(null);
    }
}

File: src/main/kotlin/com/github/thaynarasilvapinto/simuladorbanco/controller/ClienteController.kt
package com.github.thaynarasilvapinto.simuladorbanco.controller


import com.github.thaynarasilvapinto.simuladorbanco.api.ClienteApi
import com.github.thaynarasilvapinto.simuladorbanco.api.request.ClienteCriarRequest
import com.github.thaynarasilvapinto.simuladorbanco.api.response.ClienteResponse
import com.github.thaynarasilvapinto.simuladorbanco.controller.utils.toResponse
import com.github.thaynarasilvapinto.simuladorbanco.domain.Cliente
import com.github.thaynarasilvapinto.simuladorbanco.domain.Conta
import com.github.thaynarasilvapinto.simuladorbanco.services.ClienteService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@RestController
open class ClienteController(
    private val serviceCliente: ClienteService
) : ClienteApi {

    override fun mostrarCliente(@PathVariable("id") id: String): ResponseEntity<ClienteResponse> {
        val cliente = serviceCliente.cliente(id)
        return ResponseEntity.ok().body(cliente.toResponse())

    }

    override fun criarCliente(@Valid @RequestBody clienteCriarRequest: ClienteCriarRequest): ResponseEntity<ClienteResponse> {
        val cliente = serviceCliente.criarCliente(
            Cliente(
                nome = clienteCriarRequest.nome!!,
                cpf = clienteCriarRequest.cpf!!,
                conta = Conta(saldo = 0.00)
            )
        )
        return ResponseEntity.ok().body(cliente.toResponse())

    }
}